launch/launch/substitutions/equals_substitution.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the EqualsSubstitution substitution."""
+
+from typing import Any
+from typing import Iterable
+from typing import Optional
+from typing import Text
+from typing import Union
+
+from ..frontend import expose_substitution
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import is_some_substitutions_type
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+from ..utilities import normalize_to_list_of_substitutions
+from ..utilities.type_utils import perform_substitutions
+
+
+@expose_substitution('equals')
+class EqualsSubstitution(Substitution):
+    """Substitution that checks if two inputs are equal."""
+
+    def __init__(
+        self,
+        left: Optional[Union[Any, Iterable[Any]]],
```

Comment:
I'm curios why this isn't just:

```suggestion
        left: SomeSubstitutionsType,
```

Same for the below.